packages/calypso-products/src/types.ts

```diff
@@ -62,6 +63,18 @@ export type FeatureList = {
 /**
  * WPCOM
  */
+
+/**
+ * !WARNING! The following type is suspicious, how there is a reference to an Add-On slug that
+ * links to a defined feature slug. Add-Ons have been defined to refer internally to a set of
+ * features via their `featuresSlugs` prop. So `WPCOM_STORAGE_ADD_ONS` being a list of feature slugs
+ * used to individually refer to actual Add-Ons is likely a mistake.
+ */
+const WPCOM_STORAGE_ADD_ONS = < const >[
+	FEATURE_50GB_STORAGE_ADD_ON,
+	FEATURE_100GB_STORAGE_ADD_ON,
+];
+
```

Comment:
@jeyip & @southp I vaguely recall you both agreed/worked on this type right? Can you please confirm the sentiment of the comment/warning I'm leaving in the code - does it resonate?

Reply:
and if anything comes to mind about "not using feature slugs to refer to add-ons" (I think this is the main concern that resonates with me)

For more context, I've been trying to define a `FeatureSlug` type and came across this constant being exported from `features.ts`, which obviously looked like being in the wrong place. I recall some earlier convos about this. How did we resolve on it?

Reply:
> For more context, I've been trying to define a FeatureSlug type and came across this constant being exported from features.ts, which obviously looked like being in the wrong place. I recall some earlier convos about this. How did we resolve on it?

I vaguely remember this conversation 🤔  I'm going to refresh myself on the code and see where we landed.

Reply:
> For more context, I've been trying to define a FeatureSlug type and came across this constant being exported from features.ts, which obviously looked like being in the wrong place.

Confirming this now 👀. At first glance, I believe you're correct and I'm currently looking at how its used in the codebase.

Reply:
Thanks @jeyip for digging into this so deeply and rigorously! I certainly know Spiderman! 😆 

Let me revisit all of that you wrote and confirm. but everything resonates with me on first glance, especially about "removing" those references and renaming the slugs a bit to distinguish better. Let me poke around a bit and get back. :-)

Reply:
p.s. let me know what you think. I can make some quick updates/renamings in this PR too (maybe not too involving things though)

Reply:
> Hmm my immediate confusion stems from WPComStorageAddOnSlug being a reference to Feature slugs, which complicates the semantics a little for me / it feels like an indirection. Perhaps if we could rename that to WPComStorageAddOnFeatureSlug instead might be a quick win to add clarity? 🤔

That new name, at least for me, helps clarify more. I'm not sure if it would help newer devs, but it feels better than what we have right now 👍  

> Do they show up as "active features" on a site once they are purchased i.e. coming from /sites/[site]/features endpoint? If not, then there's always going to be some ambiguity/confusion if the rest of the feature slugs in calypso-products are, technically, "site features". 🤔

Really good question -- I'm not sure, but I can check in a little bit. I would intuitively assume that every product would be required to have a set of features it provides, and as a result, would always have info returned from the features endpoint. I get thrown for a loop when I see all of the client-side definitions for features in the Calypso products package though...

I think, more than anything, this is revealing how unclear this area of code is for me, because, now that you've mentioned the endpoint, I'm not even certain where the source of truth for features is anymore 😵‍💫 I'm assuming it's from the endpoint. I'll have to do some learning when I have more bandwidth.

> ^ Perhaps what we need is to highlight these as "special non-site features", somehow, and avoid including (or omitting) from a perspective FeatureSlug type that can be used to index what's returned from /sites/[site]/features (cause that's what I've been meaning to do here)

To make sure we're on the same page, could you elaborate on the term "special non-site features", specifically what would distinguish a non-site feature from a site feature? I'm assuming it just means features returned from the /sites/[site]/features endpoint.

> I am not sure. It makes sense to have individual product slugs for these add-ons - however, those will also need to map to what we have on the backend. It's been another source of hurdles getting the product slugs aligned - see this issue I created recently (definitely worth of P2 post and PT around IMO): https://github.com/Automattic/wp-calypso/issues/87826 So if we have a "product" on the frontend that's not a "product" on the backend, then that's obviously a source of ambiguity (same with what I mentioned about "features" above, so not that different?)

I see what you're saying 🤔 I'll take a closer look at your P2 soon.

> p.s. let me know what you think. I can make some quick updates/renamings in this PR too (maybe not too involving things though)

Our discussions seem like they'll require more thought and deliberation, so in the meantime, I think it's totally fine to make the quick changes. I'm going to think over this situation more.

Reply:
Thanks @jeyip for thinking this through with me.

> To make sure we're on the same page, could you elaborate on the term "special non-site features", specifically what would distinguish a non-site feature from a site feature? I'm assuming it just means features returned from the /sites/[site]/features endpoint.

You are raising a good point with this. After all, what are "features" really and why should they be restricted to being "site/plan features"?

I think going forward, we could investigate all these feature slugs (constants) in [features.ts](https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-products/src/constants/features.ts) and what they are referring to i.e. have we been strictly referring to "site/plan features" so far and is that ok going forward? Conceptually, products could have features. Can I import everything from `features.ts` and define a `SiteFeatureSlug` to index the `useSiteFeatures` return in this PR? etc.

Needs more drilling.